app/controllers/v0/facilities/va_controller.rb

```diff
@@ -12,10 +12,32 @@ class V0::Facilities::VaController < FacilitiesController
   # @param type - Optional facility type, values = all (default), health, benefits, cemetery
   # @param services - Optional specialty services filter
   def index
-    resource = BaseFacility.query(params).paginate(page: params[:page], per_page: BaseFacility.per_page)
-    render json: resource,
-           each_serializer: VAFacilitySerializer,
-           meta: metadata(resource)
+    if BaseFacility::TYPES.include?(params[:type]) || params[:address].nil?
```

Comment:
could the block of code inside the first condition also be a method? If so you could use guard clauses:
```ruby
facilities if BaseFacility::TYPES.include?(params[:type]) || params[:address].nil?
provider_locator if params[:type] == 'cc_provider'
combined
```

Reply:
didn't work quite as written. I had to put return statements before the first two conditions to avoid calling two methods every call.